packages/content/types/index.d.ts

```diff
@@ -55,9 +57,7 @@ declare module '@nuxt/types' {
       yaml?: yamlOptions,
       csv?: csvOptions,
       xml?: xmlOptions,
-      extendParser?: {
-        [extension: string]: (file: string) => any
-      }
+      extendParser?: parsers<string> | (defaultParsers: parsers<'.md'|'.yml'|'.yaml'|'.json'|'.json5'|'.xml'>) => parsers<string>
```

Comment:
Can we allow users to extend the `parsers<'.md'|'.yml'|'.yaml'|'.json'|'.json5'|'.xml'>` list with their own extensions? As to extend the list to match their config, maybe by passing a generic parameter?

Reply:
The default parsers have known extensions so no need to extend this.

Reply:
This entire PR is to allow users to define their own extensions. So I mean to let them add to this list.

Reply:
These are the default parsers so only for files with extension md, yml, yaml, json, json5, xml. The user can not change what he gets 

See in Code:
https://github.com/nuxt/content/blob/95d2712de1f5e86e56c77c21e08a108df596f2a7/packages/content/lib/database.js#L26-L34
https://github.com/nuxt/content/blob/95d2712de1f5e86e56c77c21e08a108df596f2a7/packages/content/lib/database.js#L44

Reply:
Oh missed csv.

Reply:
Oh my bad, it's the argument passed to the user of course... Nevermind then, looks good 👍